Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs

```diff
@@ -359,35 +360,50 @@ void traceCopyFailed(Context c)
                                 Tracer.Warning(
                                     context,
                                     $"{AttemptTracePrefix(attemptCount)} Could not copy file with hash {hashInfo.ContentHash} from path {sourcePath} to temp path {tempLocation} due to an internal error: {copyFileResult} Not trying another replica.");
-                                _contentLocationStore.MachineReputationTracker.ReportReputation(location, MachineReputation.Bad);
+                                _contentLocationStore.ReportReputation(location, MachineReputation.Bad);
                                 break;
                             default:
-                                return (new ErrorResult(copyFileResult, $"{AttemptTracePrefix(attemptCount)} File copier result code {copyFileResult.Code} is not recognized").AsResult<PutResult>(), true);
+                                return (result: new ErrorResult(copyFileResult, $"{AttemptTracePrefix(attemptCount)} File copier result code {copyFileResult.Code} is not recognized").AsResult<PutResult>(), retry: true);
                         }
 
                         if (copyFileResult.Succeeded)
                         {
+                            // The copy succeeded, but it is possible that the resulting size doesn't match an expected one.
+                            if (hashInfo.Size != -1 && copyFileResult.Size != null && hashInfo.Size != copyFileResult.Size.Value)

```

Comment:
Given the weird size values we have seen in the past, I would be slightly more careful in this test. Suggest running only if hashInfo.Size > 0.